### PR TITLE
[proxylib] Empty memcache rules allow all traffic

### PR DESCRIPTION
**Pull request by nebril**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5885)

### CODE REVIEW

**Comment by nebril** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

File: proxylib/proxylib_memcached_test.go [hunk 1]
```diff
@@ -572,6 +572,23 @@ var textTestCases = []testCase{
 			}, proxylib.OK, "")
 		},
 	},
+	{
+		"text set pass on empty rule",
+		`		        rule: <
+		          key: "keyExact"
+		          value: ""
+				>
```
Comment (jrajahalme) [inline proxylib/proxylib_memcached_test.go]: You should make the rule really empty ("").

Comment (jrajahalme) [reply proxylib/proxylib_memcached_test.go]: Also, does it make sense to allow empty string as the "keyExact" value? Or is it the same as no key?

File: proxylib/memcached/parser.go [hunk 2]
```diff
@@ -130,7 +136,11 @@ func L7RuleParser(rule *cilium.PortNetworkPolicyRule) []proxylib.L7NetworkPolicy
 			}
 		}
 		if !commandFound {
-			proxylib.ParseError("command not specified", rule)
+			if len(br.keyExact) > 0 || len(br.keyPrefix) > 0 || br.regex != nil {
+				proxylib.ParseError("command not specified but key was provided", rule)
+			} else {
+				br.empty = true
+			}
```
Comment (jrajahalme) [inline proxylib/memcached/parser.go:142]: While looking at this I noticed potential problem with `commandFound`. Currently it is set regardless the result of the map lookup. Should it be set like this instead?
```
			switch k {
			case "command":
				br.commands, commandFound = MemcacheOpCodeMap[v]
```

**Review by jrajahalme — DISMISSED** [review]

File: proxylib/memcached/parser.go (hunk 2 above)
Comment (nebril) [reply proxylib/memcached/parser.go]: This would actually panic without commandFound, but we can also do it your way.

**Comment by nebril** [comment]
test-me-please

File: proxylib/proxylib_memcached_test.go (hunk 1 above)
Comment (nebril) [reply proxylib/proxylib_memcached_test.go]: done

**Comment by nebril** [comment]
test-me-please

**Review by jrajahalme — APPROVED** [review]

**Pull request merged**